javascript/src/test/rationals/Rational.test.js

```diff
@@ -61,29 +61,52 @@ describe("Rational class tests", () => {
 
   describe("sum/plus tests", () => {
     test("(1 pts) zero identity tests", () => {
-      expect(one.plus(zero)).toEqual(one);
-      expect(zero.plus(one)).toEqual(one);
+      const onePlusZero = one.plus(zero);
+      const zeroPlusOne = zero.plus(one);
+
+      expect(onePlusZero).toEqual(one);
+      expect(zeroPlusOne).toEqual(one);
+
+      // must return a new instance
+      expect(Object.is(onePlusZero, one)).toBe(false);
+      expect(Object.is(zeroPlusOne, one)).toBe(false);
+
+      // must not mutate inputs
+      expect(one).toMatchObject({ numerator: 1, denominator: 1});
+      expect(zero).toMatchObject({ numerator: 0, denominator: 1});
     });
 
     test("(1 pts) 1/3 + 2/3 = 1/1", () => {
       const expected = new Rational(1, 1);
       const actual = oneThird.plus(twoThirds);
 
       expect(actual).toMatchObject(expected);
+
+      // must not mutate inputs
+      expect(oneThird).toMatchObject({ numerator: 1, denominator: 3 });
+      expect(twoThirds).toMatchObject({ numerator: 2, denominator: 3 });
     });
 
     test("(1 pts) 1/1 + -1/1 = 0", () => {
       const expected = new Rational(0, 1);
       const actual = one.plus(negativeOne);
 
       expect(actual).toMatchObject(expected);
+
+      // must not mutate inputs
+      expect(one).toMatchObject({ numerator: 1, denominator: 1 });
+      expect(negativeOne).toMatchObject({ numerator: -1, denominator: 1 });
     });
 
     test("(1 pts) -1/1 + 1/1 = 0", () => {
       const expected = new Rational(0, 1);
       const actual = negativeOne.plus(one);
 
       expect(actual).toMatchObject(expected);
+
+      // must not mutate inputs
+      expect(one).toMatchObject({ numerator: 1, denominator: 1 });
+      expect(negativeOne).toMatchObject({ numerator: -1, denominator: 1 });
     });
 
     test("(1 pts) sum gives same result as plus", () => {
```

Comment:
Is there a reason you decided not to check mutation of the inputs on these types of tests? (sum/plus, times/multiply, etc)

Reply:
I figured that it would be covered by the other tests, though we could also add tests here to be thorough

Reply:
I think it'd be more consistent to have them